sql/tests: TestRandomSyntaxGeneration failed [SHOW STATISTICS parse/format does not round-trip]

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11429805?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11429805?buildTab=artifacts#/) on release-22.2 @ [42fa1fad9af04139575e4f374a8b3f2d4ca453b1](https://github.com/cockroachdb/cockroach/commits/42fa1fad9af04139575e4f374a8b3f2d4ca453b1):


```
    rsg_test.go:822: 1m15s of 5m0s: 274436 executions, 14465 successful
    rsg_test.go:822: 1m20s of 5m0s: 292720 executions, 15384 successful
    rsg_test.go:822: 1m25s of 5m0s: 311388 executions, 16312 successful
    rsg_test.go:822: 1m30s of 5m0s: 329786 executions, 17249 successful
    rsg_test.go:822: 1m35s of 5m0s: 348910 executions, 18188 successful
    rsg_test.go:822: 1m40s of 5m0s: 366079 executions, 19019 successful
    rsg_test.go:822: 1m45s of 5m0s: 385412 executions, 19942 successful
    rsg_test.go:822: 1m50s of 5m0s: 404341 executions, 20813 successful
    rsg_test.go:822: 1m55s of 5m0s: 424190 executions, 21680 successful
    rsg_test.go:822: 2m0s of 5m0s: 441887 executions, 22490 successful
    rsg_test.go:822: 2m5s of 5m0s: 459642 executions, 23290 successful
    rsg_test.go:822: 2m10s of 5m0s: 477008 executions, 24125 successful
    rsg_test.go:822: 2m15s of 5m0s: 495276 executions, 25007 successful
    rsg_test.go:822: 2m20s of 5m0s: 513291 executions, 25856 successful
    rsg_test.go:822: 2m25s of 5m0s: 531821 executions, 26832 successful
    rsg_test.go:822: 2m30s of 5m0s: 550399 executions, 27613 successful
    rsg_test.go:822: 2m35s of 5m0s: 567933 executions, 28424 successful
    rsg_test.go:822: 2m40s of 5m0s: 586497 executions, 29272 successful
    rsg_test.go:822: 2m45s of 5m0s: 605264 executions, 30091 successful
    rsg_test.go:822: 2m50s of 5m0s: 623500 executions, 30930 successful
    rsg_test.go:822: 2m55s of 5m0s: 641132 executions, 31733 successful
    rsg_test.go:822: 3m0s of 5m0s: 659592 executions, 32562 successful
    rsg_test.go:822: 3m5s of 5m0s: 677688 executions, 33388 successful
    rsg_test.go:822: 3m10s of 5m0s: 695416 executions, 34204 successful
    rsg_test.go:822: 3m15s of 5m0s: 714206 executions, 35050 successful
    rsg_test.go:822: 3m20s of 5m0s: 733431 executions, 35917 successful
    rsg_test.go:822: 3m25s of 5m0s: 751380 executions, 36748 successful
    rsg_test.go:822: 3m30s of 5m0s: 770502 executions, 37636 successful
    rsg_test.go:822: 3m35s of 5m0s: 789003 executions, 38516 successful
    rsg_test.go:822: 3m40s of 5m0s: 806558 executions, 39317 successful
    rsg_test.go:822: 3m45s of 5m0s: 825492 executions, 40169 successful
    rsg_test.go:822: 3m50s of 5m0s: 843354 executions, 41035 successful
    rsg_test.go:822: 3m55s of 5m0s: 860534 executions, 41832 successful
    rsg_test.go:822: 4m0s of 5m0s: 878013 executions, 42670 successful
    rsg_test.go:822: 4m5s of 5m0s: 896131 executions, 43521 successful
    rsg_test.go:822: 4m10s of 5m0s: 913824 executions, 44350 successful
    rsg_test.go:822: 4m15s of 5m0s: 933603 executions, 45164 successful
    rsg_test.go:822: 4m20s of 5m0s: 952462 executions, 46043 successful
    rsg_test.go:822: 4m25s of 5m0s: 971762 executions, 46928 successful
    rsg_test.go:822: 4m30s of 5m0s: 990611 executions, 47786 successful
    rsg_test.go:822: 4m35s of 5m0s: 1008808 executions, 48646 successful
    rsg_test.go:822: 4m40s of 5m0s: 1027589 executions, 49496 successful
    rsg_test.go:822: 4m45s of 5m0s: 1046870 executions, 50346 successful
    rsg_test.go:822: 4m50s of 5m0s: 1065939 executions, 51244 successful
    rsg_test.go:822: 4m55s of 5m0s: 1082936 executions, 52024 successful
    rsg_test.go:858: 1101897 executions, 52819 successful
    rsg_test.go:866: cannot parse output of Format: sql="SHOW STATISTICS FOR TABLE FAMILY WITH OPTIONS ( BUCKET_COUNT , SETTING , 'string' , 'string' = LOW , 'string' )", formattedSQL="SHOW STATISTICS FOR TABLE \"family\" WITH bucket_count, setting, \"string\", \"string\" = 'low', \"string\"": at or near "with": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration4008816221
--- FAIL: TestRandomSyntaxGeneration (301.42s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109090 sql/tests: TestRandomSyntaxGeneration failed [descriptor not found during BACKUP write-job-info-insert] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery X-noreuse branch-master]
- #108083 sql/tests: TestRandomSyntaxGeneration failed [index visibility precision] [C-test-failure O-robot T-sql-queries X-noreuse branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30826

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11485022?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11485022?buildTab=artifacts#/) on release-22.2 @ [a3cfc62a12e650ae8cf957b66eeda3ec37177337](https://github.com/cockroachdb/cockroach/commits/a3cfc62a12e650ae8cf957b66eeda3ec37177337):


```
    rsg_test.go:822: 1m15s of 5m0s: 263184 executions, 13944 successful
    rsg_test.go:822: 1m20s of 5m0s: 282703 executions, 14805 successful
    rsg_test.go:822: 1m25s of 5m0s: 301645 executions, 15676 successful
    rsg_test.go:822: 1m30s of 5m0s: 319518 executions, 16524 successful
    rsg_test.go:822: 1m35s of 5m0s: 338341 executions, 17407 successful
    rsg_test.go:822: 1m40s of 5m0s: 356872 executions, 18268 successful
    rsg_test.go:822: 1m45s of 5m0s: 375542 executions, 19100 successful
    rsg_test.go:822: 1m50s of 5m0s: 393204 executions, 19942 successful
    rsg_test.go:822: 1m55s of 5m0s: 411516 executions, 20794 successful
    rsg_test.go:822: 2m0s of 5m0s: 428690 executions, 21525 successful
    rsg_test.go:822: 2m5s of 5m0s: 445295 executions, 22286 successful
    rsg_test.go:822: 2m10s of 5m0s: 462716 executions, 23092 successful
    rsg_test.go:822: 2m15s of 5m0s: 479342 executions, 23851 successful
    rsg_test.go:822: 2m20s of 5m0s: 497508 executions, 24621 successful
    rsg_test.go:822: 2m25s of 5m0s: 515677 executions, 25465 successful
    rsg_test.go:822: 2m30s of 5m0s: 533248 executions, 26293 successful
    rsg_test.go:822: 2m35s of 5m0s: 551465 executions, 27121 successful
    rsg_test.go:822: 2m40s of 5m0s: 570201 executions, 27921 successful
    rsg_test.go:822: 2m45s of 5m0s: 586919 executions, 28716 successful
    rsg_test.go:822: 2m50s of 5m0s: 605106 executions, 29560 successful
    rsg_test.go:822: 2m55s of 5m0s: 622891 executions, 30330 successful
    rsg_test.go:822: 3m0s of 5m0s: 642151 executions, 31171 successful
    rsg_test.go:822: 3m5s of 5m0s: 660110 executions, 31937 successful
    rsg_test.go:822: 3m10s of 5m0s: 679072 executions, 32801 successful
    rsg_test.go:822: 3m15s of 5m0s: 696895 executions, 33614 successful
    rsg_test.go:822: 3m20s of 5m0s: 715494 executions, 34446 successful
    rsg_test.go:822: 3m25s of 5m0s: 733881 executions, 35219 successful
    rsg_test.go:822: 3m30s of 5m0s: 751119 executions, 36036 successful
    rsg_test.go:822: 3m35s of 5m0s: 769145 executions, 36850 successful
    rsg_test.go:822: 3m40s of 5m0s: 787438 executions, 37670 successful
    rsg_test.go:822: 3m45s of 5m0s: 805740 executions, 38444 successful
    rsg_test.go:822: 3m50s of 5m0s: 823696 executions, 39275 successful
    rsg_test.go:822: 3m55s of 5m0s: 841159 executions, 40099 successful
    rsg_test.go:822: 4m0s of 5m0s: 857617 executions, 40858 successful
    rsg_test.go:822: 4m5s of 5m0s: 875222 executions, 41604 successful
    rsg_test.go:822: 4m10s of 5m0s: 893277 executions, 42449 successful
    rsg_test.go:822: 4m15s of 5m0s: 911843 executions, 43282 successful
    rsg_test.go:822: 4m20s of 5m0s: 929209 executions, 44054 successful
    rsg_test.go:822: 4m25s of 5m0s: 947319 executions, 44872 successful
    rsg_test.go:822: 4m30s of 5m0s: 965006 executions, 45692 successful
    rsg_test.go:822: 4m35s of 5m0s: 982418 executions, 46467 successful
    rsg_test.go:822: 4m40s of 5m0s: 1000912 executions, 47332 successful
    rsg_test.go:822: 4m45s of 5m0s: 1019150 executions, 48200 successful
    rsg_test.go:822: 4m50s of 5m0s: 1037314 executions, 49027 successful
    rsg_test.go:822: 4m55s of 5m0s: 1055130 executions, 49839 successful
    rsg_test.go:858: 1072028 executions, 50631 successful
    rsg_test.go:866: cannot parse output of Format: sql="EXPERIMENTAL CHANGEFEED FOR TABLE FAMILY FAMILY VARBIT , FAMILY . CROSS . ident FAMILY TIMESTAMP WITH OPTIONS ( BUCKET_COUNT = 'string' , 'string' , 'string' = 'string' )", formattedSQL="EXPERIMENTAL CHANGEFEED FOR TABLE \"family\" FAMILY \"varbit\", TABLE \"family\".cross.ident FAMILY \"timestamp\" WITH bucket_count = 'string', \"string\", \"string\" = 'string'": at or near "with": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration799138182
--- FAIL: TestRandomSyntaxGeneration (301.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #108083 sql/tests: TestRandomSyntaxGeneration failed [index visibility precision] [C-test-failure O-robot T-sql-queries X-noreuse branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -703,7 +703,7 @@ func TestBackupAndRestoreJobDescription(t *testing.T) {
 	sqlDB.Exec(t, "BACKUP TO ($1,$2,$3) INCREMENTAL FROM $4", append(incrementals, backups[0])...)
 	sqlDB.Exec(t, "BACKUP INTO ($1, $2, $3)", collections...)
 	sqlDB.Exec(t, "BACKUP INTO LATEST IN ($1, $2, $3)", collections...)
-	sqlDB.Exec(t, "BACKUP INTO LATEST IN ($1, $2, $3) WITH incremental_location = ($4, $5, $6)",
+	sqlDB.Exec(t, "BACKUP INTO LATEST IN ($1, $2, $3) WITH OPTIONS (incremental_location = ($4, $5, $6))",
 		append(collections, incrementals...)...)
 
 	sqlDB.ExpectErr(t, "the incremental_location option must contain the same number of locality",

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -4348,15 +4348,15 @@ func TestChangefeedDescription(t *testing.T) {
 	}{
 		{
 			create: "CREATE CHANGEFEED FOR foo INTO $1 WITH updated, envelope = $2",
-			descr:  `CREATE CHANGEFEED FOR TABLE foo INTO '` + redactedSink + `' WITH envelope = 'wrapped', updated`,
+			descr:  `CREATE CHANGEFEED FOR TABLE foo INTO '` + redactedSink + `' WITH OPTIONS (envelope = 'wrapped', updated)`,
 		},
 		{
 			create: "CREATE CHANGEFEED FOR public.foo INTO $1 WITH updated, envelope = $2",
-			descr:  `CREATE CHANGEFEED FOR TABLE public.foo INTO '` + redactedSink + `' WITH envelope = 'wrapped', updated`,
+			descr:  `CREATE CHANGEFEED FOR TABLE public.foo INTO '` + redactedSink + `' WITH OPTIONS (envelope = 'wrapped', updated)`,
 		},
 		{
 			create: "CREATE CHANGEFEED FOR d.public.foo INTO $1 WITH updated, envelope = $2",
-			descr:  `CREATE CHANGEFEED FOR TABLE d.public.foo INTO '` + redactedSink + `' WITH envelope = 'wrapped', updated`,
+			descr:  `CREATE CHANGEFEED FOR TABLE d.public.foo INTO '` + redactedSink + `' WITH OPTIONS (envelope = 'wrapped', updated)`,
 		},
 		{
 			// TODO(#85143): remove schema_change_policy='stop' from this test.

--- a/pkg/ccl/changefeedccl/show_changefeed_jobs_test.go
+++ b/pkg/ccl/changefeedccl/show_changefeed_jobs_test.go
@@ -418,7 +418,7 @@ func TestShowChangefeedJobsAlterChangefeed(t *testing.T) {
 		out = obtainJobRowFn()
 
 		require.Equal(t, jobID, out.id, "Expected id:%d but found id:%d", jobID, out.id)
-		require.Equal(t, "CREATE CHANGEFEED FOR TABLE d.public.bar INTO 'kafka://does.not.matter/' WITH resolved = '5s'", out.description, "Expected description:%s but found description:%s", "CREATE CHANGEFEED FOR TABLE bar INTO 'kafka://does.not.matter/ WITH resolved = '5s''", out.description)
+		require.Equal(t, "CREATE CHANGEFEED FOR TABLE d.public.bar INTO 'kafka://does.not.matter/' WITH OPTIONS (resolved = '5s')", out.description, "Expected description:%s but found description:%s", "CREATE CHANGEFEED FOR TABLE bar INTO 'kafka://does.not.matter/ WITH resolved = '5s''", out.description)
 		require.Equal(t, sinkURI, out.SinkURI, "Expected sinkUri:%s but found sinkUri:%s", sinkURI, out.SinkURI)
 		require.Equal(t, "bar", out.topics, "Expected topics:%s but found topics:%s", "bar", sortedTopics)
 		require.Equal(t, "{d.public.bar}", string(out.FullTableNames), "Expected fullTableNames:%s but found fullTableNames:%s", "{d.public.bar}", string(out.FullTableNames))

--- a/pkg/sql/importer/import_csv_mark_redaction_test.go
+++ b/pkg/sql/importer/import_csv_mark_redaction_test.go
@@ -30,7 +30,7 @@ func TestMarkRedactionCCLStatement(t *testing.T) {
 		expected string
 	}{
 		{
-			"IMPORT CSV 'file' WITH delimiter = 'foo'",
+			"IMPORT CSV 'file' WITH OPTIONS (delimiter = 'foo')",
 			"IMPORT CSV ‹'file'› WITH OPTIONS (delimiter = ‹'foo'›)",
 		},
 	}

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -2281,7 +2281,7 @@ func TestImportCSVStmt(t *testing.T) {
 			`CREATE TABLE t (a int8 primary key, b string, index (b), index (a, b))`,
 			`IMPORT INTO t CSV DATA (%s) WITH decompress = 'none'`,
 			testFiles.gzipFiles,
-			` WITH decompress = 'none'`,
+			` WITH OPTIONS (decompress = 'none')`,
 			// This returns different errors for `make test` and `make testrace` but
 			// field is in both error messages.
 			`field`,
@@ -2291,7 +2291,7 @@ func TestImportCSVStmt(t *testing.T) {
 			`CREATE TABLE t (a int8 primary key, b string, index (b), index (a, b))`,
 			`IMPORT INTO t CSV DATA (%s) WITH decompress = 'gzip'`,
 			testFiles.files,
-			` WITH decompress = 'gzip'`,
+			` WITH OPTIONS (decompress = 'gzip')`,
 			"gzip: invalid header",
 		},
 		{
@@ -3123,7 +3123,7 @@ func TestImportIntoCSV(t *testing.T) {
 			"import-into-no-decompress-gzip",
 			`IMPORT INTO t (a, b) CSV DATA (%s) WITH decompress = 'none'`,
 			testFiles.gzipFiles,
-			` WITH decompress = 'none'`,
+			` WITH OPTIONS (decompress = 'none')`,
 			// This returns different errors for `make test` and `make testrace` but
 			// field is in both error messages.
 			"field",
@@ -3132,21 +3132,21 @@ func TestImportIntoCSV(t *testing.T) {
 			"import-into-no-decompress-gzip",
 			`IMPORT INTO t (a, b) CSV DATA (%s) WITH decompress = 'gzip'`,
 			testFiles.files,
-			` WITH decompress = 'gzip'`,
+			` WITH OPTIONS (decompress = 'gzip')`,
 			"gzip: invalid header",
 		},
 		{
 			"import-no-files-match-wildcard",
 			`IMPORT INTO t (a, b) CSV DATA (%s) WITH decompress = 'auto'`,
 			[]string{`'nodelocal://0/data-[0-9][0-9]*'`},
-			` WITH decompress = 'auto'`,
+			` WITH OPTIONS (decompress = 'auto')`,
 			`pq: no files matched`,
 		},
 		{
 			"import-into-no-glob-wildcard",
 			`IMPORT INTO t (a, b) CSV DATA (%s) WITH disable_glob_matching`,
 			testFiles.filesUsingWildcard,
-			` WITH disable_glob_matching`,
+			` WITH OPTIONS (disable_glob_matching)`,
 			"pq: (.+)no such file or directory: nodelocal storage file does not exist:",
 		},
 	} {

--- a/pkg/sql/parser/testdata/changefeed
+++ b/pkg/sql/parser/testdata/changefeed
@@ -23,6 +23,14 @@ EXPLAIN CREATE CHANGEFEED FOR TABLE foo INTO '_' -- literals removed
 EXPLAIN CREATE CHANGEFEED FOR TABLE _ INTO 'sink' -- identifiers removed
 
 parse
+EXPERIMENTAL CHANGEFEED FOR TABLE foo WITH OPTIONS (BUCKET_COUNT = 'string', 'string')
+----
+EXPERIMENTAL CHANGEFEED FOR TABLE foo WITH OPTIONS (bucket_count = 'string', "string") -- normalized!
+EXPERIMENTAL CHANGEFEED FOR TABLE (foo) WITH OPTIONS (bucket_count = ('string'), "string") -- fully parenthesized
+EXPERIMENTAL CHANGEFEED FOR TABLE foo WITH OPTIONS (bucket_count = '_', "string") -- literals removed
+EXPERIMENTAL CHANGEFEED FOR TABLE _ WITH OPTIONS (_ = 'string', _) -- identifiers removed
+
+parse
 CREATE CHANGEFEED FOR foo INTO 'sink'
 ----
 CREATE CHANGEFEED FOR TABLE foo INTO 'sink' -- normalized!
@@ -62,10 +70,10 @@ CREATE CHANGEFEED FOR TABLE _ INTO 'sink' -- identifiers removed
 parse
 CREATE CHANGEFEED FOR TABLE foo INTO 'sink' WITH bar = 'baz'
 ----
-CREATE CHANGEFEED FOR TABLE foo INTO 'sink' WITH bar = 'baz'
-CREATE CHANGEFEED FOR TABLE (foo) INTO ('sink') WITH bar = ('baz') -- fully parenthesized
-CREATE CHANGEFEED FOR TABLE foo INTO '_' WITH bar = '_' -- literals removed
-CREATE CHANGEFEED FOR TABLE _ INTO 'sink' WITH _ = 'baz' -- identifiers removed
+CREATE CHANGEFEED FOR TABLE foo INTO 'sink' WITH OPTIONS (bar = 'baz') -- normalized!
+CREATE CHANGEFEED FOR TABLE (foo) INTO ('sink') WITH OPTIONS (bar = ('baz')) -- fully parenthesized
+CREATE CHANGEFEED FOR TABLE foo INTO '_' WITH OPTIONS (bar = '_') -- literals removed
+CREATE CHANGEFEED FOR TABLE _ INTO 'sink' WITH OPTIONS (_ = 'baz') -- identifiers removed
 
 parse
 CREATE CHANGEFEED AS SELECT * FROM foo

--- a/pkg/sql/parser/testdata/show
+++ b/pkg/sql/parser/testdata/show
@@ -263,6 +263,14 @@ SHOW PUBLIC CLUSTER SETTINGS FOR TENANT $1 -- literals removed
 SHOW PUBLIC CLUSTER SETTINGS FOR TENANT $1 -- identifiers removed
 
 parse
+SHOW PUBLIC CLUSTER SETTINGS FOR TENANT INTERVAL 'string' DAY TO HOUR
+----
+SHOW PUBLIC CLUSTER SETTINGS FOR TENANT ('string'::INTERVAL DAY TO HOUR) -- normalized!
+SHOW PUBLIC CLUSTER SETTINGS FOR TENANT ((('string')::INTERVAL DAY TO HOUR)) -- fully parenthesized
+SHOW PUBLIC CLUSTER SETTINGS FOR TENANT ('_'::INTERVAL DAY TO HOUR) -- literals removed
+SHOW PUBLIC CLUSTER SETTINGS FOR TENANT ('string'::INTERVAL DAY TO HOUR) -- identifiers removed
+
+parse
 SHOW CLUSTER SETTING a FOR TENANT INTERVAL 'string' MONTH
 ----
 SHOW CLUSTER SETTING a FOR TENANT ('string'::INTERVAL MONTH) -- normalized!
@@ -977,6 +985,14 @@ SHOW STATISTICS FOR TABLE d.t
 SHOW STATISTICS FOR TABLE d.t -- fully parenthesized
 SHOW STATISTICS FOR TABLE d.t -- literals removed
 SHOW STATISTICS FOR TABLE _._ -- identifiers removed
+
+parse
+SHOW STATISTICS FOR TABLE d.t WITH OPTIONS (BUCKET_COUNT, SETTING, 'string', 'string' = LOW)
+----
+SHOW STATISTICS FOR TABLE d.t WITH OPTIONS (bucket_count, setting, "string", "string" = 'low') -- normalized!
+SHOW STATISTICS FOR TABLE d.t WITH OPTIONS (bucket_count, setting, "string", "string" = ('low')) -- fully parenthesized
+SHOW STATISTICS FOR TABLE d.t WITH OPTIONS (bucket_count, setting, "string", "string" = '_') -- literals removed
+SHOW STATISTICS FOR TABLE _._ WITH OPTIONS (_, _, _, _ = 'low') -- identifiers removed
 
 parse
 SHOW HISTOGRAM 123

--- a/pkg/sql/sem/tree/changefeed.go
+++ b/pkg/sql/sem/tree/changefeed.go
@@ -47,8 +47,9 @@ func (node *CreateChangefeed) Format(ctx *FmtCtx) {
 		ctx.FormatNode(node.SinkURI)
 	}
 	if node.Options != nil {
-		ctx.WriteString(" WITH ")
+		ctx.WriteString(" WITH OPTIONS (")
 		ctx.FormatNode(&node.Options)
+		ctx.WriteString(")")
 	}
 }
 

--- a/pkg/sql/sem/tree/show.go
+++ b/pkg/sql/sem/tree/show.go
@@ -766,8 +766,9 @@ func (node *ShowTableStats) Format(ctx *FmtCtx) {
 	ctx.WriteString("FOR TABLE ")
 	ctx.FormatNode(node.Table)
 	if len(node.Options) > 0 {
-		ctx.WriteString(" WITH ")
+		ctx.WriteString(" WITH OPTIONS (")
 		ctx.FormatNode(&node.Options)
+		ctx.WriteString(")")
 	}
 }
 

